### PR TITLE
[fix] older versions of the ss command do not have the --no-header option

### PR DESCRIPTION
**Pull request by jyf111**
Related to #226 and #214.

Solution: execute the `ss` command in a temporary container to precheck for ports in use instead.

Side effects: the port precheck process will now take a bit longer, since we need to pull image, create a container, start the container, exec the `ss` command for each port, and then stop and remove the container.

### CODE REVIEW

File: internal/task/task/checker/network.go [hunk 1]
```diff
@@ -82,19 +130,31 @@ func NewCheckPortInUseTask(curveadm *cli.CurveAdm, dc *topology.DeployConfig) (*
...
+	t.AddStep(&step.PullImage{
+		Image:       dc.GetContainerImage(),
+		ExecOptions: curveadm.ExecOptions(),
+	})
+
+	var containerId string
+	ss_template := "ss"
+	ss_template += " --listening"
+	ss_template += " --no-header"
+	ss_template += " --tcp '" + FORMAT_FILTER_SPORT + "'"
```
Comment (Wine93) [inline internal/task/task/checker/network.go:142]:
```suggestion
cli := ctx.Module().Shell().SocketStatistics("")
cli.AddOption("--no-header")
cli.AddOption("--processes")
cli.AddOption("--listening")
command, err := cli.String()
if err != nil {
    return nil, err
}
```

Comment (jyf111) [reply internal/task/task/checker/network.go]: `CreateContainer` is wrapped by `step2InitCheckPort`, then use ctx.Module().Shell().SocketStatistics.

**Review by caoxianfei1 — CHANGES_REQUESTED** [review]

**Comment by caoxianfei1** [comment]
lgtm

File: internal/task/task/checker/network.go [hunk 2]
```diff
@@ -82,19 +130,36 @@ func NewCheckPortInUseTask(curveadm *cli.CurveAdm, dc *topology.DeployConfig) (*
...
+	var containerId, out string
 	var success bool
+	command := "-c 'sleep infinity'" // keep the container running
+	t.AddStep(&step.PullImage{
+		Image:       dc.GetContainerImage(),
+		ExecOptions: curveadm.ExecOptions(),
+	})
+	t.AddStep(&step.CreateContainer{
+		Image:       dc.GetContainerImage(),
+		Command:     command,
```
Comment (Wine93) [inline internal/task/task/checker/network.go:142]: Maybe we can assgin command option directly here.

**Review by caoxianfei1 — APPROVED** [review]

**Pull request merged**